src/Task/Ant.php

```diff
@@ -36,7 +37,9 @@ public static function getConfigurableOptions(): ConfigOptionsResolver
 
     public function canRunInContext(ContextInterface $context): bool
     {
-        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+        return $context instanceof GitPreCommitContext
```

Comment:
By default, this will run the task in both pre-commit and pre-push scenario?
I think that might be a bit too much for a lot of people.

Would it make sense to have a new metadata field that can be used to change the context instead?
See https://github.com/phpro/grumphp/blob/v2.x/doc/tasks.md#metadata

Something like:

```yaml
grumphp:
    tasks:
        anytask:
            metadata:
                git_stage: ["pre-commit", "pre-push"]
```

Which can be used to specify at what stages in your git flow you want to run this task.
The canRunInContext will then by based on the metadata configuration, which will default to `["pre-commit"]`.

WDYT?